[Bug]: `TestDataSyncService/TestStartStop` unit test is not stable

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: bcd6865
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

`TestDataSyncService/TestStartStop` unit test is not stable
Test case may quit before `NotifyCheckpointUpdated` invoked

### Expected Behavior

Unit test shall pass stably

### Steps To Reproduce

_No response_

### Milvus Log

```
2023-12-18T08:37:02.1558423Z     mock_msgstream.go:534: PASS:	AsConsumer(string,string,string,string)
2023-12-18T08:37:02.1558908Z     mock_msgstream.go:534: PASS:	Chan()
2023-12-18T08:37:02.1559253Z     mock_msgstream.go:534: PASS:	Close()
2023-12-18T08:37:02.1560897Z     mock_factory.go:277: PASS:	NewTtMsgStream(string)
2023-12-18T08:37:02.1561918Z     mock_mananger.go:392: PASS:	Register(string,string,string,string,string,string)
2023-12-18T08:37:02.1564272Z [2023/12/18 08:37:02.155 +00:00] [INFO] [msgdispatcher/manager.go:163] ["dispatcherManager exited"] [role=datanode] [nodeID=1] [pchannel=by-dev-rootcoord-dml-5137921632568935412]
2023-12-18T08:37:02.1601069Z     mock_mananger.go:392: PASS:	BufferData(string,string,string,string,string)
2023-12-18T08:37:02.1602093Z     mock_mananger.go:392: PASS:	GetCheckpoint(string)
2023-12-18T08:37:02.1603012Z     mock_mananger.go:392: FAIL:	NotifyCheckpointUpdated(string,uint64)
2023-12-18T08:37:02.1605313Z         		at: [/go/src/github.com/milvus-io/milvus/internal/datanode/writebuffer/mock_mananger.go:271 /go/src/github.com/milvus-io/milvus/internal/datanode/data_sync_service_test.go:502]
2023-12-18T08:37:02.1607273Z     mock_mananger.go:392: FAIL: 3 out of 4 expectation(s) were met.
2023-12-18T08:37:02.1608183Z         	The code you are testing needs to make 1 more call(s).
2023-12-18T08:37:02.1610782Z         	at: [/go/src/github.com/milvus-io/milvus/internal/datanode/writebuffer/mock_mananger.go:392 /usr/local/go/src/testing/testing.go:1150 /usr/local/go/src/testing/testing.go:1328 /usr/local/go/src/testing/testing.go:1570]
2023-12-18T08:37:02.1620161Z     mock_broker.go:650: PASS:	UpdateSegmentStatistics(string,string)
2023-12-18T08:37:02.1621832Z     mock_broker.go:650: PASS:	GetSegmentInfo(string,string)
2023-12-18T08:37:02.1623241Z     mock_broker.go:650: PASS:	UpdateChannelCheckpoint(string,string,string)
2023-12-18T08:37:02.1624657Z --- FAIL: TestDataSyncService (0.06s)
2023-12-18T08:37:02.1625622Z     --- FAIL: TestDataSyncService/TestStartStop (0.03s)
```

### Anything else?

https://github.com/milvus-io/milvus/actions/runs/7244961964/job/19735394815?pr=29273

--- a/internal/datanode/data_sync_service_test.go
+++ b/internal/datanode/data_sync_service_test.go
@@ -499,7 +499,7 @@ func (s *DataSyncServiceSuite) TestStartStop() {
 
 	s.wbManager.EXPECT().BufferData(insertChannelName, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
 	s.wbManager.EXPECT().GetCheckpoint(insertChannelName).Return(&msgpb.MsgPosition{Timestamp: msgTs}, true, nil)
-	s.wbManager.EXPECT().NotifyCheckpointUpdated(insertChannelName, msgTs).Return()
+	s.wbManager.EXPECT().NotifyCheckpointUpdated(insertChannelName, msgTs).Return().Maybe()
 
 	ch := make(chan struct{})
 

